apps/files/lib/Command/ListFiles.php

```diff
@@ -0,0 +1,289 @@
+			->addOption("type", "", InputArgument::OPTIONAL, "Filter by type like application, image, video etc")
+			->addOption(
+				"minSize",
+				"0",
+				InputArgument::OPTIONAL,
+				"Filter by min size"
+			)
+			->addOption(
+				"maxSize",
+				"0",
+				InputArgument::OPTIONAL,
+				"Filter by max size"
+			)
+			->addOption(
+				"sort",
+				"name",
+				InputArgument::OPTIONAL,
+				"Sort by name, path, size, owner, type, perm, created-at"
+			)
+			->addOption("order", "ASC", InputArgument::OPTIONAL, "Order is either ASC or DESC");
+	}
+
+	private function getNodeInfo(Node $node): array {
+		$nodeInfo = [
+			"name" => $node->getName(),
+			"size" => \OCP\Util::humanFileSize($node->getSize()),
+			"realSize" => $node->getSize(),
+			"perm" => $node->getPermissions(),
+			"owner" => $node->getOwner()?->getDisplayName(),
+			"created-at" => $node->getCreationTime(),
+			"type" => $node->getMimePart(),
+		];
+		if($node->getMimetype() == FileInfo::MIMETYPE_FOLDER) {
+			$nodeInfo['type'] = 'directory';
+		}
+
+		return $nodeInfo;
+	}
+
+	protected function listFiles(
+		string $user,
+		OutputInterface $output,
+		?string $path = "",
+		?string $type = "",
+		?int $minSize = 0,
+		?int $maxSize = 0
+	): void {
+		try {
+			$userFolder = $this->rootFolder->getUserFolder($user);
+			/** @var Folder $pathList **/
+			$pathList = $userFolder->get('/' . $path);
+
+			$files = $pathList->getDirectoryListing();
+			foreach ($files as $file) {
+				/** @var Node $fileNode */
+				$fileNode = $file;
+				$includeType = $includeMin = $includeMax = true;
+				$nodeInfo = $this->getNodeInfo($fileNode);
+				if ($type != "" && $type !== $nodeInfo['type']) {
+					$includeType = false;
+				}
+				if ($minSize > 0) {
+					$includeMin = $file->getSize() >= $minSize;
+				}
+				if ($maxSize > 0) {
+					$includeMax = $file->getSize() <= $maxSize;
+				}
+				if ($file instanceof File) {
+					if ($includeType && $includeMin && $includeMax) {
+						$this->fileInfo[] = $nodeInfo;
+					}
+				} elseif ($file instanceof Folder) {
+					if ($includeType && $includeMin && $includeMax) {
+						$this->dirInfo[] = $nodeInfo;
+					}
+				}
+			}
+		} catch (ForbiddenException $e) {
+			$output->writeln(
+				"<error>Home storage for user $user not writable or 'files' subdirectory missing</error>"
+			);
+			$output->writeln("  " . $e->getMessage());
+			$output->writeln(
+				'Make sure you\'re running the list command only as the user the web server runs as'
+			);
+		} catch (InterruptedException $e) {
+			# exit the function if ctrl-c has been pressed
+			$output->writeln("Interrupted by user");
+		} catch (NotFoundException $e) {
+			$output->writeln(
+				"<error>Path not found: " . $e->getMessage() . "</error>"
+			);
+		} catch (\Exception $e) {
+			$output->writeln(
+				"<error>Exception during list: " . $e->getMessage() . "</error>"
+			);
+			$output->writeln("<error>" . $e->getTraceAsString() . "</error>");
+		}
```

Comment:
```suggestion
		}
```
You can simply let unexpected exceptions bubble up, parent class takes care of showing the trace depending on verbosity if I recall correctly.